node/tests/SharedTests.ts

```diff
@@ -2527,6 +2527,130 @@ export function runBaseTests<Context>(config: {
         },
         config.timeout,
     );
+
+    // Set command tests
+
+    async function classicSet(client: BaseClient, key: string, value: string) {
+        const setRes = await client.set(key, value);
+        expect(setRes).toEqual("OK");
+        const getRes = await client.get(key);
+        expect(getRes).toEqual(value);
+    }
+
+    async function setWithExpiryOptions(
+        client: BaseClient,
+        key: string,
+        value: string,
+    ) {
+        const setRes = await client.set(key, value, {
+            expiry: {
+                type: "milliseconds",
+                count: 500,
+            },
+        });
+        expect(setRes).toEqual("OK");
+        const getRes = await client.get(key);
+        expect(getRes).toEqual(value);
+
+        const setRes2 = await client.set(key, value, {
+            expiry: {
+                type: "seconds",
+                count: 1,
+            },
+        });
+        expect(setRes2).toEqual("OK");
+        const getRes2 = await client.get(key);
+        expect(getRes2).toEqual(value);
+
+        const setRes3 = await client.set(key, value, {
+            expiry: "keepExisting",
+        });
+        expect(setRes3).toEqual("OK");
+        const getRes3 = await client.get(key);
+        expect(getRes3).toEqual(value);
+
+        const setRes4 = await client.set(key, value, {
+            expiry: {
+                type: "unixMilliseconds",
+                count: 2,
+            },
+        });
+        expect(setRes4).toEqual("OK");
+        // wait for the key to expire
+        setTimeout(() => {}, 5);
+        const getRes4 = await client.get(key);
+        // key should have expired
+        expect(getRes4).toEqual(null);
+    }
+
+    async function setWithExistOptions(
+        client: BaseClient,
+        key: string,
+        value: string,
+    ) {
+        const setRes = await client.set(key, value, {
+            conditionalSet: "onlyIfExists",
+        });
+        expect(setRes).toEqual("OK");
+        const getRes = await client.get(key);
+        expect(getRes).toEqual(value);
+
+        const setRes2 = await client.set(key, value + "1", {
+            conditionalSet: "onlyIfDoesNotExist",
+        });
+        // key already exists, so it should not be set
+        expect(setRes2).toEqual(null);
+        // key was already set, so it should not be changed
+        const getRes2 = await client.get(key);
+        expect(getRes2).toEqual(value);
+    }
+
+    async function setWithGetOldOptions(
+        client: BaseClient,
+        key: string,
+        value: string,
+    ) {
+        const setRes = await client.set(key, value, {
+            returnOldValue: true,
+        });
+        expect(setRes).toEqual(value);
+        const getRes = await client.get(key);
+        expect(getRes).toEqual(value);
+        const setRes2 = await client.set(key + "1", value, {
```

Comment:
consider naming variables more verbosely
```suggestion
        const setResNonExistingKey = await client.set(key + "1", value, {
```